.github/workflows/deploy-optimized.yml

```
@@ -0,0 +1,557 @@
+    if: |
+      always() &&
+      needs.validate-deployment.outputs.validation_passed == 'true' &&
+      needs.detect-changes.outputs.migration_needed == 'true'
+    outputs:
+      migration_status: ${{ steps.migrations.outputs.status }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: 🚀 Run Migrations
+        id: migrations
+        run: |
```

actionlint (1.7.7)

property "deployment_url" is not defined in object type {health_status: string; validation_passed: string}

(expression)